### PR TITLE
IRQ_STEER portability fixes

### PR DESCRIPTION
**Pull request by paulstelian97**
On the i.MX8M platform the register map for the IRQ_STEER driver has significant differences from i.MX8/i.MX8X platforms.

This commit separates the register maps into a separate header file for each of the platforms.

### CODE REVIEW

File: src/drivers/imx/interrupt.c [hunk 1]
```diff
@@ -70,54 +70,24 @@
...
- * The interrupts are mapped in the registers in the following way:
- * Interrupts 480-511 at offset 0
- * Interrupts 448-479 at offset 1
- * Interrupts 416-447 at offset 2
- * ...
- * Interrupts 64-95 at offset 13
- * Interrupts 32-63 at offset 14
- * Interrupts 0-31 at offset 15
+ * The interrupt mapping to registers is defined in platform specific
+ * headers.
```
Comment (dbaluta) [inline src/drivers/imx/interrupt.c]: is defined in interrupt.h platform specific header? Is this more accurate?

File: src/platform/imx8/include/platform/drivers/interrupt.h [hunk 2]
```diff
@@ -77,6 +77,39 @@
...
+
+#define IRQSTR_CHANCTL			0x00
+#define IRQSTR_CH_MASK(n)		(0x04 + 0x04 * (15 - (n)))
+#define IRQSTR_CH_SET(n)		(0x44 + 0x04 * (15 - (n)))
+#define IRQSTR_CH_STATUS		(0x84 + 0x04 * (15 - (n)))
+#define IRQSTR_MASTER_DISABLE		0xC4
+#define IRQSTR_MASTER_STATUS		0xC8
+
+#define IRQSTR_RESERVED_IRQS_NUM	32
+#define IRQSTR_IRQS_NUM			512
```
Comment (dbaluta) [inline src/platform/imx8/include/platform/drivers/interrupt.h]: indentation?

File: src/platform/imx8m/include/platform/drivers/interrupt.h [hunk 3]
```diff
@@ -61,8 +61,7 @@
 /* 32 HW interrupts + 8 IRQ_STEER lines each with 64 interrupts */
 #define PLATFORM_IRQ_HW_NUM	XCHAL_NUM_INTERRUPTS
 #define PLATFORM_IRQ_CHILDREN	64 /* Each cascaded struct covers 64 IRQs */
-/* IMX: Covered steer IRQs are modulo-64 aligned. */
-#define PLATFORM_IRQ_FIRST_CHILD  0
+#define PLATFORM_IRQ_FIRST_CHILD  PLATFORM_IRQ_HW_NUM
```
Comment (dbaluta) [inline src/platform/imx8m/include/platform/drivers/interrupt.h]: Is this to be placed in a separate patch?

File: src/platform/imx8m/include/platform/drivers/interrupt.h [hunk 4]
```diff
@@ -77,6 +76,35 @@
...
+#define IRQSTR_CHANCTL			0x00
+
+#define IRQSTR_CH_MASK(n)		(0x04 + 0x04 * (n))
+#define IRQSTR_CH_SET(n)		(0x18 + 0x04 * (n))
+#define IRQSTR_CH_STATUS(n)		(0x2C + 0x04 * (n))
+#define IRQSTR_MASTER_DISABLE		0x40
+#define IRQSTR_MASTER_STATUS		0x44
+
+#define IRQSTR_RESERVED_IRQS_NUM	0
+#define IRQSTR_IRQS_NUM			160
```
Comment (dbaluta) [inline src/platform/imx8m/include/platform/drivers/interrupt.h]: IRQSTR_IRQS_NUM indentation looks strange. Or is it the github's fault?

File: src/platform/imx8/include/platform/drivers/interrupt.h (hunk 2 above)
Comment (paulstelian97) [reply src/platform/imx8/include/platform/drivers/interrupt.h]: Diff format "+ " makes tabs no longer line up. That leads to misleading indentation on Github.

File: src/platform/imx8m/include/platform/drivers/interrupt.h (hunk 3 above)
Comment (paulstelian97) [reply src/platform/imx8m/include/platform/drivers/interrupt.h]: I think I should add a patch specifically dedicated to enabling the 8MP. But that would make an intermediate version which doesn't compile.

File: src/platform/imx8m/include/platform/drivers/interrupt.h (hunk 4 above)
Comment (paulstelian97) [reply src/platform/imx8m/include/platform/drivers/interrupt.h]: Same explanation as in the other case, the "+ " from diff format makes tabs no longer line up.

File: src/drivers/imx/interrupt.c (hunk 1 above)
Comment (paulstelian97) [reply src/drivers/imx/interrupt.c]: Is it better now?

File: src/drivers/imx/interrupt.c [hunk 5]
```diff
@@ -159,11 +129,10 @@ static void irqstr_disable_hw(void)
...
-	/* First 32 interrupts are reserved, we just give 0 */
-	if (!index)
+	if (index < IRQSTR_RESERVED_IRQS_NUM / 32)
 		return 0;
 	/* On out of range for our platform, be silent */
-	if (index > 15)
+	if (index >= IRQSTR_IRQS_REGISTERS_NUM)
 		return 0;
 	return irqstr_read(IRQSTR_CH_STATUS(index));
 }
```
Comment (dbaluta) [inline src/drivers/imx/interrupt.c]: @could you move this to a separate patch? It makes code easier to read.

File: src/platform/imx8m/include/platform/drivers/interrupt.h (hunk 3 above)
Comment (dbaluta) [reply src/platform/imx8m/include/platform/drivers/interrupt.h]: We could keep this as it is in the patches that does the separation. Meaning that the code for 8MP will compile but will not work. And then next patch.

Or you could leave it in this patch but add a comment to explain why IRQ_FIRST_CHILD is no longer 0.

File: src/drivers/imx/interrupt.c (hunk 5 above)
Comment (paulstelian97) [reply src/drivers/imx/interrupt.c]: Done.

File: src/platform/imx8m/include/platform/drivers/interrupt.h (hunk 3 above)
Comment (paulstelian97) [reply src/platform/imx8m/include/platform/drivers/interrupt.h]: I feel like there may be a better way to place the comment, for example why on the other platform it is different (AFAIK only i.MX8/i.MX8X has it different from PLATFORM_IRQ_HW_NUM).

**Comment by dbaluta** [comment]
@paul build fails for imx8 care to have a look?

**Comment by paulstelian97** [comment]
> paul build fails for imx8 care to have a look?

8 and 8x were both broken due to a typo. Fixed.

**Comment by paulstelian97** [comment]
Current issues seem unrelated.

File: src/drivers/imx/interrupt.c [hunk 6]
```diff
@@ -159,11 +159,10 @@ static void irqstr_disable_hw(void)
  */
```
Comment (dbaluta) [inline src/drivers/imx/interrupt.c]: @paulstelian97 PR looks good to me. Tested on i.MX8QXP works fine. 

Care to enhance to commit message to explain why the bounds check are no longer correct for i.MX8M. Just a short summary for someone who is not familiar with inner details of irqsteer.

File: src/drivers/imx/interrupt.c [hunk 7]
```diff
@@ -70,54 +70,24 @@
  * usage in this driver.
  */
```
Comment (dbaluta) [inline src/drivers/imx/interrupt.c]: @paulstelian97 really appreciate if you can add two lines about the differences between irqsteer on 8/8X vs 8M

Comment (dbaluta) [reply src/drivers/imx/interrupt.c]: Later edit: notice that you can read this from the comments inside code.

**Comment by lgirdwood** [comment]
Jenkins CI reports are CI issues (not PR issues). @wwittbrx internal CI looks like unrelated issues ?
